trezorlib/tests/device_tests/test_msg_nem_signtx_mosaics_t2.py

```diff
@@ -168,33 +169,38 @@ def test_nem_signtx_mosaic_creation_levy(self):
 
     def _nem_sign(self, num_of_swipes, test_suite):
         n = parse_path("m/44'/1'/0'/0'/0'")
-        msg = nem.create_sign_tx(test_suite)
-        assert msg.transaction is not None
-        msg.transaction.address_n = n
-
-        # Sending NEMSignTx message
-        self.client.transport.write(msg)
-        ret = self.client.transport.read()
-
-        # Confirm action
-        assert isinstance(ret, proto.ButtonRequest)
-        self.client.debug.press_yes()
-        self.client.transport.write(proto.ButtonAck())
-        time.sleep(1)
-        for i in range(num_of_swipes):
-            self.client.debug.swipe_down()
+
+        def input_flow():
+            # Confirm Action
+            btn_code = yield
+            assert btn_code == B.ConfirmOutput
+            self.client.debug.press_yes()
+
+            # Swipe and confirm
             time.sleep(1)
-        self.client.debug.press_yes()
-        ret = self.client.transport.read()
-
-        # Confirm action
-        assert isinstance(ret, proto.ButtonRequest)
-        self.client.debug.press_yes()
-        self.client.transport.write(proto.ButtonAck())
-        ret = self.client.transport.read()
-
-        # Confirm tx
-        assert isinstance(ret, proto.ButtonRequest)
-        self.client.debug.press_yes()
-        self.client.transport.write(proto.ButtonAck())
-        return self.client.transport.read()
+            for i in range(num_of_swipes):
```

Comment:
one very minor thing that I missed:

if the `i` is unused in the loop, please use `_` for the variable, as in:
```python
for _ in range(num_of_swipes):
	do.stuff()
```